src/NServiceBus.Core/Serializers/XML/XmlMessageSerializer.cs

```diff
@@ -70,9 +70,10 @@ public object[] Deserialize(Stream stream, IList<Type> messageTypesToDeserialize
         /// </summary>
         public void Serialize(object message, Stream stream)
         {
-            var serializer = new Serializer(mapper, conventions, cache, SkipWrappingRawXml, Namespace);
-            var buffer = serializer.Serialize(message);
-            stream.Write(buffer, 0, buffer.Length);
+            using (var serializer = new Serializer(mapper, stream, message, conventions, cache, SkipWrappingRawXml, Namespace))
+            {
+                serializer.Serialize();
```

Comment:
for a public re-usable API i would probably agree with you. but in this case it is a single use internal class. and the implementation is simplified by not having to pass the writer to every method